datalad/core/local/status.py

```diff
@@ -347,15 +347,15 @@ def __call__(
                     yield dict(
                         action='status',
                         path=p,
-                        refds=ds.path,
+                        refds=ds_path,
                         status='error',
                         message='path not underneath this dataset',
                         logger=lgr)
                     continue
                 else:
                     if dataset and root == str(p) and \
                             not (orig_path.endswith(op.sep) or
-                                 orig_path == "."):
+                                 root == ds_path):
```

Comment:
and no `op.normpath` or alike needed, and something like `./.` or `subdir/..` etc would work too?  would be nice to add to the tests then.

Reply:
Good point.  At the moment, it depends on whether it's something `resolve_path` adjusts (and, as its docstring notes, that avoids some normalization that can be problematic for the reasons discussed in pathlib's documentation).

Given the comparison is with `Dataset().path`, I think the best approach is to just feed root through `Dataset()`.  That way there should be no need to match logic or keep it in sync.

Will push an update to condition/tests.